### ISSUE TITLE
Windows Kernel debugger doesn't properly pause execution

### ISSUE THREAD

**Issue by 0xFDFDFDFD** [Vector35/debugger#598]
The Windows Kernel  debugger doesn't properly pause execution.
Because when we pause the execution and we execute the `r` command twice without unpausing, the content of the registers changes:
![Screenshot_20240717_122731](https://github.com/user-attachments/assets/6d824a1e-e248-42bb-a0c6-392a111095d5)

The expected behavior should be like this in windbg:
![Screenshot_20240717_122930](https://github.com/user-attachments/assets/974d6f51-4996-4561-a018-74b529611d38)
After breaking none of the values are changing.

**Comment by xusheng6** [comment]
Thx for letting me know about this bug, I will fix it ASAP

**Comment by xusheng6** [comment]
Per my testing, the target is properly stopped -- I tried to interact with the VM and the guest system hangs. Also, it seems only the first time when you run "r", you get a different value, the subsequent values are all the same. I will look into it further but this may not be a bug

**Comment by 0xFDFDFDFD** [comment]
Yes it is only the first time i get a different value. Also the RIP changes from nt!DbgBreakPointWithStatus to nt!HalProcessorIdle where it stays.  This behavior only happens in the binary ninja debugger not when i debug the kernel with windbg. And its not only the `r` command also commands like `dd @r8` or any other command are changing the state.

**Comment by xusheng6** [comment]
Right, there is definitely something unusual going on, and I need to figure that out

**Comment by xusheng6** [comment]
This may be related to https://github.com/Vector35/debugger/issues/591
